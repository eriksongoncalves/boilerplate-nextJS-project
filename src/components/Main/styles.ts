import styled, { css } from 'styled-components'

import customMedia from 'styles/media-query'
import { fadeIn } from 'styles/animates'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    width: 100%;
    height: 100%;
    padding: 0 ${theme.spacings.medium};
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    animation: ${fadeIn} ${theme.transition.slow};
    transition: all ${theme.transition.default};

    ${customMedia.greaterThan('large')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `}
`
