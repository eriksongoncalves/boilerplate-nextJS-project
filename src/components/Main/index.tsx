import * as S from './styles';

export default function Main({ title = 'Hello World' }) {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
    </S.Wrapper>
  );
}
