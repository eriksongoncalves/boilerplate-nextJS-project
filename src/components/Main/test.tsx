import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Main from '.';

describe('<Main />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Main />);

    expect(
      screen.getByRole('heading', { name: /Hello World/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a Main', () => {
    const { container } = renderWithTheme(<Main />);

    expect(container.firstChild).toHaveStyle({ backgroundColor: '#06092b' });
  });
});
