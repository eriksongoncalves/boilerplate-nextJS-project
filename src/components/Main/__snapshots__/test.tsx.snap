// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correctly 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2.5rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Hello World
  </h1>
</main>
`;
